packages/grid/_modules_/grid/hooks/root/useGridContainerProps.ts

```diff
@@ -35,13 +35,14 @@ export const useGridContainerProps = (
 
   const getVirtualRowCount = React.useCallback(() => {
     logger.debug('Calculating virtual row count.');
-    if (options.pagination) {
+    if (options.pagination && !options.autoPageSize) {
       const rowsLeft = visibleRowsCount - paginationState.page * paginationState.pageSize;
       return rowsLeft > paginationState.pageSize ? paginationState.pageSize : rowsLeft;
```

Comment:
When autoPageSize is on we don't want to use the pageSize as it's accurate to the value of the viewport pagesize so we just return visibleRowsCount